portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -87,4 +93,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   private Optional<String> getParameter(HttpServletRequest request, String name) {
     return Optional.ofNullable(request.getParameter(name));
   }
+
+  /** Returns the choice entered by the user, or -1 if the choice was invalid. */
+  private int getNumberOfComments(HttpServletRequest request, String name) {
+    // Get the input from the form.
+    String numCommentString = request.getParameter(name);
+
+    // Convert the input to an int.
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numCommentString);
+      return -1;
```

Comment:
Don't return an arbitrary value here. Throw the exception if there is an exception. Otherwise the callers need to handle it. Although here we return -1, the exception will still be thrown from Iterables.limit.

Reply:
I took out the try catch so that it will just throw the exception; does that work?

Reply:
Yeah that's great. Can you update the javadoc as well?